core/execution/equity_shares.go

```diff
@@ -161,27 +158,40 @@ func (es *EquityShares) SetPartyStake(id string, newStakeU *num.Uint) {
 		v.stake = newStake
 		es.totalVStake = es.totalVStake.Add(v.vStake)
 		es.totalPStake = es.totalPStake.Add(v.stake)
-		v.avg = v.vStake.Mul(es.totalPStake.Div(es.totalVStake))
+		// average entry valuation should not be changed when reducing the physical stake
+		// v.avg = v.vStake.Mul(es.totalPStake.Div(es.totalVStake))
 		return
 	}
 
-	es.totalVStake = es.totalVStake.Add(delta)
-	es.totalPStake = es.totalPStake.Sub(v.stake).Add(newStake)
+	es.updateAvgPosDelta(v, delta, newStake)
+	// es.totalVStake = es.totalVStake.Add(delta)
+	// es.totalPStake = es.totalPStake.Sub(v.stake).Add(newStake)
 	// stakes were originally assigned _after_ the mustEquity call
 	// average was calculated in the if es.openingAuctionEnded bit
 	// v.avg = v.vStake.Mul(es.totalPStake.Div(es.totalVStake))
-	v.vStake = v.vStake.Add(delta) // increase
+	// v.vStake = v.vStake.Add(delta) // increase
+	// v.stake = newStake
+	// v.avg = v.vStake.Mul(es.totalPStake.Div(es.totalVStake))
```

Comment:
?

Reply:
lots of stuff was moved around to get things to match the spreadsheet and/or the spec file. Going back and forth between things ended up with a lot of lines commented out. Will clean this up